third_party/terraform/tests/resource_bigquery_table_test.go

```diff
@@ -471,12 +471,135 @@ func TestUnitBigQueryDataTable_jsonEquivalency(t *testing.T) {
 	}
 }
 
+func TestUnitBigQueryDataTable_schemaIsChangable(t *testing.T) {
+	t.Parallel()
+	for _, testcase := range testUnitBigQueryDataTableIsChangableTestCases {
+		testcase.check(t)
+		testcaseNested := &testUnitBigQueryDataTableJSONChangeableTestCase{
+			testcase.name + "Nested",
+			fmt.Sprintf("[{\"name\": \"someValue\", \"type\" : \"INTEGER\", \"fields\" : %s }]", testcase.jsonOld),
+			fmt.Sprintf("[{\"name\": \"someValue\", \"type\" : \"INT64\", \"fields\" : %s }]", testcase.jsonNew),
+			testcase.changeable,
+		}
+		testcaseNested.check(t)
+	}
+}
+
 type testUnitBigQueryDataTableJSONEquivalencyTestCase struct {
 	jsonA      string
 	jsonB      string
 	equivalent bool
 }
 
+type testUnitBigQueryDataTableJSONChangeableTestCase struct {
+	name       string
+	jsonOld    string
+	jsonNew    string
+	changeable bool
+}
+
+func (testcase *testUnitBigQueryDataTableJSONChangeableTestCase) check(t *testing.T) {
+	var old, new interface{}
+	if err := json.Unmarshal([]byte(testcase.jsonOld), &old); err != nil {
+		panic(fmt.Sprintf("unable to unmarshal json - %v", err))
+	}
+	if err := json.Unmarshal([]byte(testcase.jsonNew), &new); err != nil {
+		panic(fmt.Sprintf("unable to unmarshal json - %v", err))
+	}
+	changeable, err := resourceBigQueryTableSchemaIsChangeable(old, new)
+	if err != nil {
+		t.Errorf("ahhhh an error I did not expect this! especially not on testscase %s - %s", testcase.name, err)
```

Comment:
`t.Errorf` will keep going, which seems unexpected. Do you think we'll get new information? `t.Fatalf` will exit immediately, by contrast.

Reply:
oo this is nice to know ! Thanks :)